<a href="https://colab.research.google.com/github/kamranakhter03/Sales-and-Profit-Analysis-of-a-Retail-Store/blob/main/Exploratory_Data_Analysis_of_a_Retail_Store.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# PROJECT NAME : ```Sales and Profit Analysis of a Retail Store```

## Project Type
- #### Exploratory Data Analysis (EDA) | Business Intelligence | Data-Driven Insights

## Project Summary
- This project involved Exploratory Data Analysis (EDA) on a retail sales dataset to derive meaningful business insights. The analysis was performed on various dimensions such as product categories, sub-categories, order months, and customer segments. We identified top-performing and underperforming areas in terms of both sales and profit. November emerged as the most profitable month, while Copiers and Phones were the most profitable sub-categories. Although the Consumer segment had the highest sales, the Home Office segment showed better profit efficiency. These findings can assist in making informed decisions on inventory planning, marketing focus, and customer targeting strategies.

## GitHub Link -
https://github.com/kamranakhter03/Sales-and-Profit-Analysis-of-a-Retail-Store/tree/main

## Problem Statement
- The retail business had access to a large volume of sales and profit data, but lacked a clear understanding of how different products, time periods, and customer segments were impacting overall performance. The challenge was to explore this dataset to extract actionable insights that could help in identifying profitable areas, underperforming segments, and hidden trends in customer behavior.

## Define Your Business Objective?
- To identify which product categories generate the highest and lowest total sales.
- To determine monthly trends in total sales and highlight the most and least profitable months.
- To analyze product category and sub-category-wise contribution to overall profit.
- To understand how different customer segments (Consumer, Corporate, Home Office) contribute to sales and profit.
- To evaluate the profit efficiency of each customer segment by calculating the Sales-to-Profit ratio.
- To provide actionable insights for business decisions based on product and customer performance.
- To help optimize inventory, sales strategy, and marketing focus using EDA insights from the e-commerce data.

# *Let's Begin!*

## Step 1. Data Gathering

### Importing Required Libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import plotly.express as px              # for data visualization
import plotly.graph_objects as go        # for making and customizing advanced graph
import plotly.io as pio                  # for customizing the templates of the graph
import plotly.colors as colors           # for colors
pio.templates.default = 'plotly_white'   # default theme is white

### Loading the dataset

In [ ]:
df = pd.read_csv('SampleSuperstore.csv', encoding = 'latin-1')

## Step 2. Basic Data Exploration

In [ ]:
# Display top 5 rows
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [ ]:
# Describe the dataset

df.describe().round(2).T

,count,mean,std,min,25%,50%,75%,max
Row ID,9994.0,4997.50,2885.16,1.00,2499.25,4997.50,7495.75,9994.00
Postal Code,9994.0,55190.38,32063.69,1040.00,23223.00,56430.50,90008.00,99301.00
Sales,9994.0,229.86,623.25,0.44,17.28,54.49,209.94,22638.48
Quantity,9994.0,3.79,2.23,1.00,2.00,3.00,5.00,14.00
Discount,9994.0,0.16,0.21,0.00,0.00,0.20,0.20,0.80
Profit,9994.0,28.66,234.26,-6599.98,1.73,8.67,29.36,8399.98


In [ ]:
# Data Information

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

## Step 3. Data Preprocessing

- ### Data Cleaning

In [ ]:
# Converting date columns

df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9994 non-null   int64         
 1   Order ID       9994 non-null   object        
 2   Order Date     9994 non-null   datetime64[ns]
 3   Ship Date      9994 non-null   datetime64[ns]
 4   Ship Mode      9994 non-null   object        
 5   Customer ID    9994 non-null   object        
 6   Customer Name  9994 non-null   object        
 7   Segment        9994 non-null   object        
 8   Country        9994 non-null   object        
 9   City           9994 non-null   object        
 10  State          9994 non-null   object        
 11  Postal Code    9994 non-null   int64         
 12  Region         9994 non-null   object        
 13  Product ID     9994 non-null   object        
 14  Category       9994 non-null   object        
 15  Sub-Category   9994 n

- ### Feature Engineering

In [ ]:
df['Order Month'] = df['Order Date'].dt.month
df['Order Year'] = df['Order Date'].dt.year
df['Order Day of Week'] = df['Order Date'].dt.dayofweek

In [ ]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Order Month,Order Year,Order Day of Week
0,1,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136,11,2016,1
1,2,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820,11,2016,1
2,3,CA-2016-138688,2016-06-12,2016-06-16,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714,6,2016,6
3,4,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310,10,2015,6
4,5,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164,10,2015,6


## Step 4. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables

- ## Q.1. Monthly Sales Analysis

> You need to calculate the monthly sales of the store and identify which month had the highest sales and which month had the lowest sales

In [ ]:
sales_by_month = df.groupby('Order Month')['Sales'].sum().reset_index()
sales_by_month.sort_values(by = 'Sales', ascending=False)

,Order Month,Sales
10,11,352461.0710
11,12,325293.5035
8,9,307649.9457
2,3,205005.4888
9,10,200322.9847
7,8,159044.0630
4,5,155028.8117
5,6,152718.6793
6,7,147238.0970
3,4,137762.1286


In [ ]:
plt.figure(figsize = (10, 35))

fig = px.line(sales_by_month,
             x = 'Order Month',
             y = 'Sales',
             title = 'Monthly Sales Analysis',
             markers=True,
             color_discrete_sequence=['Royal Blue'])

fig.update_layout(title={'x': 0.5})  # x=0.5 means center
fig.show()

<Figure size 1000x3500 with 0 Axes>

### Conclusions :

After calculating the total monthly sales by grouping the data based on the 'Order Month', we found that:
- November (Order Month 11) recorded the highest sales of ₹3,52,461.07. This could be due to festive season promotions, year-end offers, or increased customer spending during holidays, which typically boost sales in this period.
- On the other hand, February (Order Month 2) had the lowest sales of ₹59,751.25, possibly because it has fewer days and less seasonal buying activity, leading to reduced revenue.

- ## Q.2. Category Wise Sales Analysis

> You need to analyze sales based on product categories and determine which category has the lowest sales and which category has the highest sales.

In [ ]:
sales_by_category = df.groupby('Category')['Sales'].sum().reset_index()
sales_by_category.sort_values(by = 'Sales', ascending=False)

,Category,Sales
2,Technology,836154.0330
0,Furniture,741999.7953
1,Office Supplies,719047.0320


In [ ]:
fig = px.pie(sales_by_category,
            values = 'Sales',
            names = 'Category',
            hole = 0.5,
            color_discrete_sequence=px.colors.qualitative.Set1
            )

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text='Sales Analysis by Category', title_font=dict(size=24))
fig.update_layout(title={'x': 0.5})
fig.show()

### Conclusions :

Based on the category-wise sales analysis,
- The Technology category recorded the highest sales of ₹8,36,154.03, mainly because technology products are high-value items and are in consistent demand, especially in business environments.
- In contrast, the Office Supplies category had the lowest sales of ₹7,19,047.03, likely due to the lower cost of items, which leads to smaller revenue despite possibly high order volumes.

- ## Q.3. Sales Analysis by Sub-Category

> The sales analysis needs to be done based on sub-categories

In [ ]:
sales_by_subcategory = df.groupby('Sub-Category')['Sales'].sum().reset_index()
sales_by_subcategory.sort_values(by = 'Sales', ascending= False)

,Sub-Category,Sales
13,Phones,330007.0540
5,Chairs,328449.1030
14,Storage,223843.6080
16,Tables,206965.5320
3,Binders,203412.7330
11,Machines,189238.6310
0,Accessories,167380.3180
6,Copiers,149528.0300
4,Bookcases,114879.9963
1,Appliances,107532.1610


In [ ]:
fig = px.bar(sales_by_subcategory,
            x = 'Sub-Category',
            y = 'Sales',
            title = 'Sales Analysis by Sub-Category',
            color_discrete_sequence=['#1f77b4']  # Elegant Blue
            )

fig.update_layout(title={'x': 0.5})  # Center the title
fig.show()

### Conclusions :

Based on sub-category-wise sales analysis,
- Phones generated the highest sales of ₹3,30,007.05. This is likely because phones are high-value products and in continuous demand for both personal and professional use.
- In contrast, Fasteners had the lowest sales of ₹3,024.28, possibly due to their low cost and limited usage, which contributes minimally to overall revenue.

- ## Q.4. Monthly Profit Analysis

> You need to analyze the monthly profit from sales and determine which month had the highest and lowest profit.

In [ ]:
profit_by_month = df.groupby('Order Month')['Profit'].sum().reset_index()
profit_by_month.sort_values(by='Profit', ascending=False)

,Order Month,Profit
11,12,43369.1919
8,9,36857.4753
10,11,35468.4265
9,10,31784.0413
2,3,28594.6872
4,5,22411.3078
7,8,21776.9384
5,6,21285.7954
6,7,13832.6648
3,4,11587.4363


In [ ]:
fig = px.line(profit_by_month,
             x = 'Order Month',
             y = 'Profit',
             title = 'Monthly Profit Analysis',
            markers=True,
            color_discrete_sequence=['#2ca02c']  # Deep Green
            )
fig.update_layout(title={'x': 0.5})
fig.show()

### Conclusions :

After analyzing the monthly profit,
- We observed that December (Order Month 12) recorded the highest profit of ₹43,369.19. This could be due to increased year-end purchases, high-margin products, or bulk corporate orders during the festive and closing season.
- On the other hand, January (Order Month 1) had the lowest profit of ₹9,134.45, possibly because it’s the post-holiday period, when customer spending typically drops and discounting might be higher, affecting profit margins.

- ## Q.5. Profit Analysis by Category and Sub-Category

> Analyze the profit by category

In [ ]:
profit_by_category = df.groupby('Category')['Profit'].sum().reset_index()
profit_by_category.sort_values(by='Profit', ascending=False)

,Category,Profit
2,Technology,145454.9481
1,Office Supplies,122490.8008
0,Furniture,18451.2728


In [ ]:
fig = px.pie(
    profit_by_category,
    values = 'Profit',
    names = 'Category',
    hole = 0.5,
    color_discrete_sequence=px.colors.qualitative.Dark2
    )

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text='Profit Analysis by Category', title_font=dict(size=24))


fig.show()

### Conclusions :

Based on profit analysis by category,
- The Technology category generated the highest profit of ₹1,45,454.95. This is because technology products often have higher profit margins and are sold at premium prices.
- In contrast, the Furniture category recorded the lowest profit of ₹18,451.27, possibly due to high shipping/storage costs, frequent discounting, or lower profit margins despite decent sales.

> Analyze the profit by sub-category

In [ ]:
profit_by_subcategory = df.groupby('Sub-Category')['Profit'].sum().reset_index()
profit_by_subcategory.sort_values(by='Profit', ascending=False)

,Sub-Category,Profit
6,Copiers,55617.8249
13,Phones,44515.7306
0,Accessories,41936.6357
12,Paper,34053.5693
3,Binders,30221.7633
5,Chairs,26590.1663
14,Storage,21278.8264
1,Appliances,18138.0054
9,Furnishings,13059.1436
7,Envelopes,6964.1767


In [ ]:
fig = px.bar(profit_by_subcategory,
            x = 'Sub-Category',
            y = 'Profit',
            title = 'Profit Analysis by Sub-Category',
            color_discrete_sequence=['#00B5B5']  # Indigo
            )
fig.update_layout(title_x=0.5)  # Center the title
fig.show()

### Conclusions :

After analyzing profit by sub-category,
- We found that Copiers generated the highest profit of ₹55,617.82. This is likely because copiers are high-ticket items with strong profit margins and less frequent discounts.
- On the other hand, Tables resulted in the highest loss of ₹-17,725.48. This may be due to high transportation/storage costs, bulk discounts, or low-margin pricing strategies, which reduced overall profitability.

- ## Q.6. Sales and Profit Analysis by Customer-Segment

> Analyze the sales and profit by customer-segment

In [ ]:
sales_by_customer_segment = df.groupby('Segment')['Sales'].sum().reset_index()
sales_by_customer_segment.sort_values(by='Sales', ascending=False)

,Segment,Sales
0,Consumer,1.161401e+06
1,Corporate,7.061464e+05
2,Home Office,4.296531e+05


In [ ]:
fig = px.pie(
    sales_by_customer_segment,
    values = 'Sales',
    names = 'Segment',
    hole = 0.5,
    color_discrete_sequence=px.colors.qualitative.Set1
)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text='Sales Analysis by Segment', title_font=dict(size=24))
fig.update_layout(title_x=0.5)  # Center the title

fig.show()

In [ ]:
profit_by_customer_segment = df.groupby('Segment')['Profit'].sum().reset_index()
profit_by_customer_segment.sort_values(by='Profit', ascending=False)

,Segment,Profit
0,Consumer,134119.2092
1,Corporate,91979.1340
2,Home Office,60298.6785


In [ ]:
fig = px.pie(
    profit_by_customer_segment,
    values = 'Profit',
    names = 'Segment',
    hole = 0.5,
    color_discrete_sequence=px.colors.qualitative.Set1
)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text='Profit Analysis by Segment', title_font=dict(size=24))

fig.show()

- #### Sales and Profit Analysis by Segment (Together)

In [ ]:
sales_profit_analysis_by_segment = df.groupby('Segment').agg({'Sales' : 'sum', 'Profit' : 'sum'}).reset_index()
sales_profit_analysis_by_segment

,Segment,Sales,Profit
0,Consumer,1.161401e+06,134119.2092
1,Corporate,7.061464e+05,91979.1340
2,Home Office,4.296531e+05,60298.6785


In [ ]:
fig = go.Figure()

fig.add_trace(go.Bar(
    x = sales_profit_analysis_by_segment['Segment'],
    y = sales_profit_analysis_by_segment['Sales'],
    name = 'Sales',
    marker_color = '#1F77B4'  # Muted Blue
))

fig.add_trace(go.Bar(
    x = sales_profit_analysis_by_segment['Segment'],
    y = sales_profit_analysis_by_segment['Profit'],
    name = 'Profit',
    marker_color = '#FF7F0E'  # Soft Orange
))

fig.update_layout(
    title='Sales and Profit Analysis by Customer Segment',
    title_x=0.5,
    xaxis_title='Customer Segment',
    yaxis_title='Amount',
    barmode='group'
)

fig.show()


### Conclusions :

From the sales and profit analysis by customer segment:
- The Consumer segment generated the highest sales of ₹11,61,401.00 and also achieved the highest profit of ₹1,34,119.21. This indicates strong demand and healthy margins in the B2C (business-to-consumer) space, likely driven by frequent individual purchases and seasonal buying trends.
- The Home Office segment recorded the lowest sales of ₹4,29,653.10 and the lowest profit of ₹60,298.68, possibly due to smaller order sizes, limited budgets, and lower frequency of purchases.

This shows that the Consumer segment is the most profitable and revenue-generating customer group, while the Home Office segment contributes the least in both aspects.

- ### Q.7. Analyze the sales to profit ratio

In [ ]:
sales_profit_by_segment = df.groupby('Segment').agg({'Sales' : 'sum', 'Profit' : 'sum'}).reset_index()
sales_profit_by_segment['Sales_to_Profit_Ratio'] = sales_profit_by_segment['Sales'] / sales_profit_by_segment['Profit']
print(sales_profit_by_segment[['Segment', 'Sales_to_Profit_Ratio']])

       Segment  Sales_to_Profit_Ratio
0     Consumer               8.659471
1    Corporate               7.677245
2  Home Office               7.125416


### Conclusions :

The Sales to Profit Ratio was calculated for each customer segment to understand how much sales were required to generate ₹1 profit.
- The Consumer segment had the highest ratio of 8.66, meaning it takes ₹8.66 in sales to earn ₹1 in profit.
- The Home Office segment had the lowest ratio of 7.13, indicating it is more efficient in converting sales into profit compared to other segments.

### Interpretation:
    
- Although the Consumer segment has the highest sales and profit in absolute terms, its higher sales-to-profit ratio suggests lower profit efficiency.
- On the other hand, Home Office customers, despite contributing the least sales, yield better profit per rupee of sales, making them a more cost-effective segment.

#  
 ************************************************************************************************************************************************************************
# 📝 Final Summary: Sales and Profit Analysis of a Retail Store
In this project, we performed a detailed Exploratory Data Analysis (EDA) on a retail store dataset to derive meaningful insights about its sales and profit performance across different dimensions. The main objective was to understand how the business is performing over time, across product lines, and among different customer segments.

## ✅ Steps Performed in the Project:
### Data Grouping and Aggregation:
- Using groupby() and aggregation functions in Python (Pandas), we calculated monthly sales, profits, and breakdowns by category, sub-category, and segment.

### Trend Analysis (Monthly):
- We analyzed monthly sales and profits to identify seasonal trends. November showed the highest sales, while February was the lowest. December brought the highest profits, and January had the least.

### Category and Sub-Category Analysis:
- We examined both sales and profit contributions of each product category and sub-category.
- Technology and Phones/Copiers were top performers in sales and profits.
- Furniture and Tables either contributed the least or led to losses.

### Segment-wise Analysis:
- We studied customer segments (Consumer, Corporate, Home Office) to analyze which type of customer brings in more revenue and profit. The Consumer segment had the highest revenue and profit, while the Home Office segment was the most efficient in terms of profit per sales.

### Sales to Profit Ratio Calculation:
- We calculated the Sales-to-Profit ratio to assess efficiency. A lower ratio means better profit generation per rupee of sales. Home Office had the lowest ratio (7.13) — indicating it’s the most cost-effective segment.

## 🛠️ Tools and Technologies Used

- **Python**: Core programming language for analysis  
- **Pandas**: Data manipulation and preprocessing  
- **NumPy**: Numerical operations  
- **Matplotlib**: Basic data visualizations  
- **Seaborn**: Statistical data visualizations  
- **Plotly Express**: Interactive and advanced plotting  
- **Plotly Graph Objects**: Custom and layered visualizations  
- **Plotly IO & Colors**: Template and color customization  
- **Google Colab**: Cloud-based notebook environment

## 📌 Key Insights:
- Festive seasons (Nov–Dec) boost both sales and profit.
- High-ticket items like Phones and Copiers generate more profit.
- Tables and Bookcases may need attention due to consistent losses.
- Customer segmentation helps identify which group is more valuable and efficient.

## ✅ Conclusion:
This project gave a complete business overview of the store’s performance. By breaking down the data month-wise, category-wise, and segment-wise, we could highlight not only what sells the most, but also what earns the most, and where the business should focus or improve.

### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***